src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -119,6 +122,8 @@ public Http2Connection(HttpConnectionPool pool, Connection connection)
             _pendingWindowUpdate = 0;
             _idleSinceTickCount = Environment.TickCount64;
 
+            _keepAlive = new Http2KeepAlive(this, _pool.Settings);
```

Comment:
If keep alive is not enabled, i.e. KeepAlivePingDelay is off, then you don't need to create this type.